File: app/src/main/java/com/amir/alzheimer/infrastructure/OptionsAdapter.java
package com.amir.alzheimer.infrastructure;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.amir.alzheimer.R;
import com.amir.alzheimer.base.BaseActivity;

public class OptionsAdapter extends RecyclerView.Adapter<OptionsViewHolder> implements View.OnClickListener {

    private static final String TAG = "OptionsAdapter";
    public static final int[] IMAGES = {
            R.mipmap.ic_doc,
            R.mipmap.ic_mind,
            R.mipmap.ic_set,
            R.mipmap.ic_note,
            R.mipmap.ic_med,
            R.mipmap.ic_rem
    };

    private final int itemSize;
    private final AlzheimerItemCallback listener;
    private LayoutInflater inflater;

    public OptionsAdapter(BaseActivity activity, int itemSize, AlzheimerItemCallback listener) {
        inflater = LayoutInflater.from(activity);
        Log.e(TAG, "OptionsAdapter: adapter was created");
        this.itemSize = itemSize;
        this.listener = listener;
    }

    @NonNull
    @Override
    public OptionsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.item_alzheimer_option, parent, false);
        ViewGroup.LayoutParams params = new ViewGroup.LayoutParams(itemSize, itemSize);
        view.setLayoutParams(params);
        view.setOnClickListener(this);
        return new OptionsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(OptionsViewHolder holder, int position) {
        holder.populate(IMAGES[position % IMAGES.length]);
      }

    @Override
    public int getItemCount() {
        return Integer.MAX_VALUE;
    }

    @Override
    public void onClick(View v) {
        int intID = (v.getTag() == null) ? -1 : (Integer) v.getTag();
        listener.itemClicked(intID);
    }
}

File: app/src/main/java/com/amir/alzheimer/infrastructure/OptionsAdapter.kt
package com.amir.alzheimer.infrastructure

import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.amir.alzheimer.R
import com.amir.alzheimer.base.BaseActivity

class OptionsAdapter(activity: BaseActivity, private val itemSize: Int, private val listener: AlzheimerItemCallback) : RecyclerView.Adapter<OptionsViewHolder>(), View.OnClickListener {
    private val tags: Array<String> = arrayOf(activity.getString(R.string.doctor),
            activity.getString(R.string.brain),
            activity.getString(R.string.settings),
            activity.getString(R.string.relatives),
            activity.getString(R.string.medicines),
            activity.getString(R.string.reminders)
    )

    private val inflater: LayoutInflater = LayoutInflater.from(activity)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OptionsViewHolder {
        val view = inflater.inflate(R.layout.item_alzheimer_option, parent, false)
        val params = ViewGroup.LayoutParams(itemSize, itemSize)
        view.layoutParams = params
        view.setOnClickListener(this)
        return OptionsViewHolder(view)
    }

    override fun onBindViewHolder(holder: OptionsViewHolder, position: Int) {
        holder.populate(IMAGES[position % IMAGES.size], tags[position % IMAGES.size])
    }

    override fun getItemCount(): Int {
        return Integer.MAX_VALUE
    }

    override fun onClick(v: View) {
        val intID = if (v.tag == null) -1 else v.tag as Int
        listener.itemClicked(intID)
    }

    companion object {

        private val TAG = "OptionsAdapter"
        val IMAGES = intArrayOf(R.mipmap.ic_doc, R.mipmap.ic_mind, R.mipmap.ic_set, R.mipmap.ic_note, R.mipmap.ic_med, R.mipmap.ic_rem)
    }
}
